src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubConfiguration.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.ServiceBus.Messaging;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using System.Globalization;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+
+namespace Microsoft.Azure.WebJobs.ServiceBus
+{
+    public class EventHubConfiguration : IExtensionConfigProvider, IEventHubProvider
+    {
+        // $$$ are names case-sensitive?
+        Dictionary<string, EventHubClient> _senders = new Dictionary<string, EventHubClient>();
+        Dictionary<string, EventProcessorHost> _listeners  = new Dictionary<string, EventProcessorHost>();
+
+        private List<Action<JobHostConfiguration>> _deferredWork = new List<Action<JobHostConfiguration>>();
+
+        public void AddSender(EventHubClient client)
+        {
+            string name = client.Path;
+            _senders[name] = client;
+        }
+
+        public void AddSender(string eventHubName, string sendConnectionString)
+        {
+            var client = EventHubClient.CreateFromConnectionString(sendConnectionString, eventHubName);
+            AddSender(client);
+        }
+
+        
+        private void AddReceiver2(string eventHubName, EventProcessorHost listener)
+        {
+            _listeners[eventHubName] = listener;
+        }
+
+        public void AddReceiver(string eventHubName, string receiverConnectionString)
+        {
+            // We can't get the storage string until we get a JobHostConfig. So defer this. 
+            _deferredWork.Add((jobHostConfig) =>
+           {
+               string storageConnectionString = jobHostConfig.DashboardConnectionString;
```

Comment:
Why does the EH receiver need the Dashboard connection string? Also you're assigning the dashboard connection to a storageConnection variable - they can be two different accounts.